### ISSUE TITLE
Swagger (ui) is not generating correct sample xml for @XmlRootElement

### ISSUE THREAD

**Issue by elonderin** [springfox/springfox#2630]
## TLDR 
1. the general models representation at the bottom of the page
2. @XmlRootElement.name is not considered in sample XML for
a) request parameters ~~~unless there is also a @JacksonXmlRootElement present~~~
b) return  types

## Steps to Reproduce
An example of the above bugs is in https://github.com/elonderin/springfox-swagger2-bug-xmlrootelement. 

This is a fork of gs-rest-service that i extended for this case with these models:
 
```java
    @JacksonXmlRootElement(localName = "greeting-jaxb")
    @XmlRootElement(name = "greeting-jaxb")
    public class Greeting2 {
    
      @XmlElement(name = "my-id")
      ...
    }
    
    @XmlRootElement(name = "greeting-jaxb")
    public class Greeting {
    
      @XmlElement(name = "my-id")
      private final long   id;
      private final String content;
      ...
    }
```

1. compile and run the complete/ subfolder
1. open http://localhost:8080/swagger-ui.html#/

### Case: the general models representation at the bottom of the page
1. open the **Models** section
:boom: both have the java class name, neither the Jason nor JAXB Annotations are honored

### Case: @XmlRootElement.name is not considered in sample XML for request parameters unless there is also a @JacksonXmlRootElement present
1. open `/post`
:boom: the root tag is `<Greeting>` but should be `<greeting-jaxb>` 
1. open `/post2`
~~~:white_check_mark: the root tag is `<greeting-jaxb>`~~~ 
:sob: now that i write this issue report, i cant reproduce this anymore, so plz ignore for now ...

### Case: @XmlRootElement.name is not considered in sample XML for return types
1. open `/greeting`
:boom: the root tag is `<Greeting>` but should be `<greeting-jaxb>` 
 
## Context
This might be related to #2393 but this bug seems to tackle another/more specific case this one is simpler 

* it  might only affect the UI
* it is tested against 2.9.2

**Comment by dilipkrish** [comment]
This can be easily implemented as a plugin that [copies JacksonXmlModelPlugin](https://github.com/springfox/springfox/blob/master/springfox-schema/src/main/java/springfox/documentation/schema/plugins/JacksonXmlModelPlugin.java)
